gensim/models/poincare_visualization.py

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/env python
```

Comment:
I think better write this code in ipython notebook (or maybe move to `docs/notebooks` and import from "under the feet").
`gensim.model` isn't a suitable place for this file.

Reply:
I agree `gensim.models` isn't a good place for it. I don't want to have it in an ipython notebook or `docs/notebooks` though since that would mean a user can't import and use it, and I think it is definitely useful for a user. Do you think creating a new package `poincare` in `gensim/` would be a good idea? Other models do this too (e.g. `topic_coherence`)

Reply:
I don't think that this is a very good idea, after refactoring, `topic_coherence` will be moved/renamed in "deep" of `gensim.modules` (coherence contains only inner/secondary functions, not public API). 

I agree about imports (it's really untrivial, how to import from `docs/notebooks` if you in `/randomfolder`, only manually with `importlib` I think.

We have many viz helpers (produced by @parulsethi on GSoC) + now Parul works on very nice with for topic models. Potentially, we can create the distinct repository (like `gensim-data`) and move all viz helpers, or, as you suggest, create submodule `gensim.viz` and move all viz stuff (not only your Poincare viz).

Hard question, I don't know what's better right now.

WDYT @piskvorky @janpom @parulsethi?

Reply:
A submodule `gensim.viz` sounds good to me, keeping in mind we might have future visualizations too. I don't have a good enough perspective on this though, so whatever you decide is okay with me.

Reply:
Hi @menshikh-iv , so is it okay if I create a `gensim.py` package and move the poincare visualization there?

Reply:
@piskvorky @parulsethi

Reply:
`gensim.viz` submodule would be useful for https://github.com/RaRe-Technologies/gensim/pull/1616 also in future, and the long code blocks of [network graph](https://github.com/RaRe-Technologies/gensim/blob/develop/docs/notebooks/topic_network.ipynb)/[dendrogram](https://github.com/RaRe-Technologies/gensim/blob/develop/docs/notebooks/Topic_dendrogram.ipynb) could also be wrapped up in a function under this module so that those visualizations can be produced simply using the imports.

Reply:
@jayantj sounds good.

Reply:
Great! Thanks for the feedback @parulsethi @menshikh-iv 
Pushed changes with a new `gensim.viz` package.